runners/spark/src/main/java/org/apache/beam/runners/spark/TestSparkRunner.java

```diff
@@ -135,11 +138,13 @@ public SparkPipelineResult run(Pipeline pipeline) {
             isOneOf(PipelineResult.State.STOPPED, PipelineResult.State.DONE));
 
         // validate assertion succeeded (at least once).
-        int successAssertions = 0;
-        try {
-          successAssertions = result.getAggregatorValue(PAssert.SUCCESS_COUNTER, Integer.class);
-        } catch (NullPointerException e) {
-          // No assertions registered will cause an NPE here.
+        long successAssertions = 0;
+        Iterable<MetricResult<Long>> counterResults = result.metrics().queryMetrics(
+            MetricsFilter.builder()
+                .addNameFilter(MetricNameFilter.named(PAssert.class, PAssert.SUCCESS_COUNTER))
+                .build()).counters();
+        for (MetricResult<Long> counter : counterResults) {
+          successAssertions += counter.attempted().longValue();
```

Comment:
SGTM. Since each `PAssert` is a separate transformation (step) it will have its own counter, so what @bjchambers suggests is an improvement on the previous assertion method, in cases of over-counting.